modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/Marshaling.cs

```diff
@@ -721,8 +721,9 @@ public static godot_callable ConvertCallableToNative(in Callable p_managed_calla
             if (p_managed_callable.Delegate != null)
             {
                 var gcHandle = CustomGCHandle.AllocStrong(p_managed_callable.Delegate);
+                ulong objectId = p_managed_callable.Target?.GetInstanceId() ?? 0;
```

Comment:
For performance, it's better to pass the object pointer to the native function and get the instance id from there:

```suggestion
                IntPtr objectPtr = p_managed_callable.Target != null ? Object.GetPtr(p_managed_callable.Target) : IntPtr.Zero;
```